### PR TITLE
Opening cfile fixes

### PR DESCRIPTION
**Pull request by achaikou**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,6 +94,8 @@ target_compile_definitions(lfp
         $<$<BOOL:${LFP_BIG_ENDIAN}>:IS_BIG_ENDIAN>
 )
 
+add_compile_definitions(ROOT_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}\")
```
Comment (jokva) [inline CMakeLists.txt]: target_compile_definitions

File: src/cfile.cpp [hunk 2]
```diff
@@ -51,8 +51,11 @@ class cfile : public lfp_protocol {
 void cfile::close() noexcept (false) {
     /*
      * The file handle will always be closed when the destructor is invoked,
-     * but when close is invoked directly, errors will be propagated
+     * but when close is invoked directly, errors will be propagated.
+     * Multiple calls are allowed: method does nothing if the file handle
```
Comment (jokva) [inline src/cfile.cpp]: In protocol.hpp instead.

File: test/cfile.cpp [hunk 3]
```diff
@@ -28,3 +28,17 @@ TEST_CASE(
...
+
+TEST_CASE(
+    "Layering non-existing file",
+    "[cfile][filehandle]") {
+
+    FILE* fp    = std::fopen(getdatafile("nonexisting.txt").c_str(), "r");
+    auto* cfile = lfp_cfile(fp);
+    auto* tif   = lfp_tapeimage_open(cfile);
+
+    CHECK(fp    == nullptr);
```
Comment (jokva) [inline test/cfile.cpp]: CHECK(not fp) or CHECK(!fp)

File: include/lfp/protocol.hpp [hunk 4]
```diff
@@ -164,6 +168,14 @@ class unique_lfp {
         this->fp.reset(nullptr);
     }
 
+    /** Implicit conversion to `bool`
+     *
+     *  Checks whether an object is owned.
+     */
+    explicit operator bool() noexcept (true) {
```
Comment (achaikou) [inline include/lfp/protocol.hpp]: Question: why if I declare this method "const" (as in unique_ptr), it suddenly stops being picked up for conversion?

Comment (jokva) [reply include/lfp/protocol.hpp]: I suppose the non-const pointer conversion is preferred, since the unique_lfp reference is non-const.

**Comment by jokva** [comment]
Looks great.

**Pull request merged**